### PR TITLE
Fix reshape translation for SCIP

### PR DESCRIPTION
**Pull request by jonathanberthias**
It works surprisingly well, though I suspect some corner cases might uncover as more test cases are added.

### CODE REVIEW

File: src/cvxpy_translation/scip/translation.py [hunk 1]
```diff
@@ -561,11 +561,15 @@ def visit_reshape(
             except AttributeError:
                 return np.reshape(x, target_shape)
         expr = self.visit(x)
```
[the remark this answers is not in the document]
Comment (jonathanberthias) [reply src/cvxpy_translation/scip/translation.py:563]: This was done on purpose, the old naming was unnecessary

File: src/cvxpy_translation/scip/interface.py [hunk 2]
```diff
@@ -177,14 +177,13 @@ def get_constraint_dual(
...
-        except Warning:
-            return None
-
-    return None
+        return np.array(model.getDualsolLinear(constr))
+
+    dual = np.zeros(shape)
+    for idx, subconstr_name in _matrix_to_scip_names(constraint_name, shape):
+        constr = get_constraint_by_name(model, subconstr_name)
+        dual[idx] = model.getDualsolLinear(constr)
```
[the remark this answers is not in the document]
Comment (jonathanberthias) [reply src/cvxpy_translation/scip/interface.py:185]: The current tests never fail, I'll see later in which cases this can raise
